Firmware/src/modules/tools/temperaturecontrol/max31855.cpp

```diff
@@ -0,0 +1,151 @@
+      Smoothie is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+      You should have received a copy of the GNU General Public License along with Smoothie. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <math.h>
+#include <fcntl.h>
+
+#include "lpc43_ssp.h"
+#include "max31855.h"
+#include "OutputStream.h"
+
+#define chip_select_pin_key "chip_select_pin"
+#define spi_channel_key         "spi_channel"
+#define designator_key           "designator"
+#define tool_id_key                 "tool_id"
+
+//Define maximum number of max31855 sensors which may be configured
+//TODO this defined number should be removed as we would like to configure any number of sensors
+#define NUM_MAX31855_SENSORS 2
+Max31855 *Max31855::instance[NUM_MAX31855_SENSORS];
+uint8_t Max31855::instance_index = 0;
+RingBuffer *Max31855::queue[NUM_MAX31855_SENSORS];
+
+/* The following functions outside class are provided for Nuttx:
+ * .select functions control GPIO Chip Select pin during data acquisition for target sensor
+ * .status functions perform status operations (not currently used but need to be here)
+ */
+
+// For enabled SSP0 channel
+void lpc43_ssp0select(FAR struct spi_dev_s *dev, enum spi_dev_e devid, bool selected)
+{
+    Max31855::instance[Max31855::instance_index]->spi_cs_pin.set(!selected);
+}
+uint8_t lpc43_ssp0status(FAR struct spi_dev_s *dev, enum spi_dev_e devid){return 0;}
+
+// For enabled SSP1 channel
+void lpc43_ssp1select(FAR struct spi_dev_s *dev, enum spi_dev_e devid, bool selected)
+{
+    Max31855::instance[Max31855::instance_index]->spi_cs_pin.set(!selected);
+}
+uint8_t lpc43_ssp1status(FAR struct spi_dev_s *dev, enum spi_dev_e devid){return 0;}
+
+/* Initialize target sensor */
+Max31855::Max31855(){
+    this->read_flag=true;
+    this->index=instance_index;
+    instance[instance_index] = this;
+    queue[instance_index++] = new RingBuffer(16); //buffer initialized with 16 elements
+}
+
+/* Configure the module using the parameters from the config file */
+bool Max31855::configure(ConfigReader& cr, ConfigReader::section_map_t& m)
+{
+    /* select which SPI channel to use:
+        0: SPI
+        1: SSP0   (default)
+        2: SSP1   */
+    this->spi_channel = cr.get_int(m, spi_channel_key,1);
+    if(spi_channel < 0 || spi_channel > 2) {
+        printf("WARNING: Invalid SPI channel %d\n",spi_channel);
+        return false;
+    }
+
+    /* open file in Nuttx corresponding to the selected SPI channel
+        /dev/max31855_0: SPI
+        /dev/max31855_1: SSP0   (default)
+        /dev/max31855_2: SSP1   */
+    std::string devpath="/dev/max31855_";
```

Comment:
nuttx does not have a /dev/max31855_*, it uses /dev/spi1 etx I think.

Reply:
I am preparing a pull request for smoothie-nuttx, which adds max31855 support for bambino-200e. The idea is to open a file on nuttx which registers max31855 driver and acquire data using the max31855 functions on nuttx. So , i open two files "max31855_1 and max31855_2", which corresponds to register max31855 on SSP0 and SSP1 ports, respectively.

Reply:
Please do not do this, it will not be accepted. Why can you not use the default /dev/spi driver? as I originally requested?
Changes to nuttx will not be accepted unless unavoidable.